From c8625f1d837ed182fa299373f047ab3034d0cc39 Mon Sep 17 00:00:00 2001
From: Samuel Verschelde <stormi@laposte.net>
Date: Wed, 5 Jun 2019 18:20:22 +0200
Subject: [PATCH] Fix installation on CoreOS

- Fix detection of recent CoreOS
- Fix installation on any version of CoreOS (script attempted to copy a
  file that does not exist anymore in the tools distribution)

Refs https://github.com/xcp-ng/xcp/issues/183

diff --git a/install.sh b/install.sh
index 1a6f0b6..a97907e 100755
--- a/install.sh
+++ b/install.sh
@@ -496,9 +496,7 @@ install_coreos()
 {
     echo "Installing the agent..."
     mkdir -p /usr/share/oem/xs/
-    cp -f ${DATADIR}/xe-daemon \
-          ${DATADIR}/xe-linux-distribution \
-          ${DATADIR}/xe-update-guest-attrs /usr/share/oem/xs/
+    cp -f ${DATADIR}/xe-daemon ${DATADIR}/xe-linux-distribution /usr/share/oem/xs/
     cp -f ${DATADIR}/xen-vcpu-hotplug.rules /etc/udev/rules.d/
     cp -f ${DATADIR}/xe-linux-distribution.service /etc/systemd/system/
     systemctl enable /etc/systemd/system/xe-linux-distribution.service
@@ -513,6 +511,7 @@ case "${os_distro}" in
     debian|ubuntu|linx|yinhe)          select_pkgs_debian ;;
     xe-ddk|xe-sdk)                     select_pkgs_xe ;;
     CoreOS)                            select_pkgs_coreos ;;
+    "Container Linux by CoreOS")       select_pkgs_coreos ;;
     *)                  failure "Unknown Linux distribution \`${os_distro}'." ;;
 esac
 

